docker-compose.yml

```diff
@@ -1,81 +1,85 @@
-version: '2.1'
-
-networks:
-  get5:
-    external: true
+version: "3.7"
 
 services:
-  caddy:
-    image: caddy:2-alpine
-    container_name: caddy-reverse-proxy
-    restart: unless-stopped
-    ports:
-      - "80:80"
-      - "443:443"
-    volumes: 
-      - ./Caddyfile:/etc/caddy/Caddyfile
+  redis:
+    image: redis:6
+    command: redis-server --requirepass testing
+    container_name: redis
     networks:
       - get5
-    environment:
-      - CADDY_URL=
-      - CADDY_API_ENDPOINT=api
-      - CADDY_REVERSE_PROXY_PORT=3301
-  
+    restart: always
+
   get5db:
     image: mariadb
     container_name: get5db
     restart: always
+    networks:
+      - get5
     environment:
-      - MYSQL_ROOT_PASSWORD=
+      - MYSQL_ROOT_PASSWORD=testing
       - MYSQL_DATABASE=get5
       - MYSQL_USER=get5
-      - MYSQL_PASSWORD=
-    ports:
-      - "3306:3306"
+      - MYSQL_PASSWORD=testing
+
+  caddy:
+    image: lucaslorentz/caddy-docker-proxy:ci-alpine
+    container_name: caddy-reverse-proxy
+    restart: unless-stopped
     networks:
       - get5
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    ports:
+      - 80:80
+      - 443:443
+    environment:
+      - CADDY_INGRESS_NETWORKS=get5
 
   g5api:
-    image: ghcr.io/phlexplexico/g5api:latest
+    build: .
     depends_on:
       - get5db
     container_name: G5API
-    volumes: 
-      - redisVol:/RedisFiles
-    expose:
-      - 3301
+    networks:
+      - get5
+    labels:
+      caddy: localhost
+      caddy.handle_path: /api/*
+      caddy.handle_path.0_reverse_proxy: "{{upstreams 3301}}"
     environment:
+      - NODE_ENV=production
       - PORT=3301
-      - HOSTNAME=
       - DBKEY=
       - STEAMAPIKEY=
-      - SHAREDSECRET=
-      - CLIENTHOME=
-      - APIURL=
+      - HOSTNAME=https://localhost
+      - SHAREDSECRET=testing
+      - CLIENTHOME=https://localhost
+      - APIURL=https://localhost/api
       - SQLUSER=get5
-      - SQLPASSWORD=
+      - SQLPASSWORD=testing
       - SQLPORT=3306
       - DATABASE=get5
       - SQLHOST=get5db
       - ADMINS=
       - SUPERADMINS=
-      - REDISPASSWORD=
+      - REDISURL=redis://:testing@redis:6379
+      - REDISTTL=86400
+      - USEREDIS=true
       - UPLOADDEMOS=true
-    restart: unless-stopped
-    networks:
-      - get5
+    restart: always
 
   g5v:
-    image: ghcr.io/phlexplexico/g5v:latest
-    depends_on: 
+    image: g5v:latest
```

Comment:
Please set to `ghcr.io/phlexplexico/g5v:latest` as we are not hosting on Docker, but on GHCR.